src/syntax_util.ml

```diff
@@ -3,7 +3,39 @@ open Asttypes
 open Parsetree
 open Longident
 
-let a = Str.global_replace
+(** [is_prefixed prefix i str] checks if prefix is the prefix of str
+  * starting from position i
+  *)
+let is_prefixed prefix i str =
+  let len = String.length prefix in
+  if i + len > String.length str then
+    1
+  else
+    String.compare (String.sub str i len) prefix
```

Comment:
There is no need to allocate a string here. Especially because `is_prefixed` is called at every step in `replace_string`.